### ISSUE TITLE
Illegal ambiguous match on configurable attribute "deps" in //tensorflow/core:ops

### ISSUE THREAD

**Issue by adamjstewart** [tensorflow/tensorflow#34878]
**System information**
- OS Platform and Distribution: macOS Catalina 10.15.1
- TensorFlow installed from: source
- TensorFlow version: 2.0.0
- Python version: 3.7.4
- Installed using: [spack](https://spack.io)
- Bazel version: 0.26.1
- Compiler version: Clang 11.0.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the problem**

I'm trying to build a minimal TF installation, but I don't seem to be able to disable GCP support. I get the following error during build time:
```
/private/var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/Adam/spack-stage/spack-stage-py-tensorflow-2.0.0-ve5t7rb2qt4igekam6hdrwnm6vgts22q/spack-src/tensorflow/core/BUILD:1359:12: Illegal ambiguous match on configurable attribute "deps" in //tensorflow/core:ops:
//tensorflow:api_version_2
//tensorflow:no_gcp_support
Multiple matches are not allowed unless one is unambiguously more specialized.
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**

First, I set the following environment variables:
```bash
TF_CONFIGURE_IOS=0; export TF_CONFIGURE_IOS
TF_DOWNLOAD_CLANG=0; export TF_DOWNLOAD_CLANG
TF_ENABLE_XLA=0; export TF_ENABLE_XLA
TF_NEED_AWS=0; export TF_NEED_AWS
TF_NEED_CUDA=0; export TF_NEED_CUDA
TF_NEED_GCP=0; export TF_NEED_GCP
TF_NEED_GDR=0; export TF_NEED_GDR
TF_NEED_HDFS=0; export TF_NEED_HDFS
TF_NEED_IGNITE=0; export TF_NEED_IGNITE
TF_NEED_JEMALLOC=0; export TF_NEED_JEMALLOC
TF_NEED_KAFKA=0; export TF_NEED_KAFKA
TF_NEED_MKL=0; export TF_NEED_MKL
TF_NEED_MPI=0; export TF_NEED_MPI
TF_NEED_NGRAPH=0; export TF_NEED_NGRAPH
TF_NEED_OPENCL=0; export TF_NEED_OPENCL
TF_NEED_OPENCL_SYCL=0; export TF_NEED_OPENCL_SYCL
TF_NEED_ROCM=0; export TF_NEED_ROCM
TF_NEED_S3=0; export TF_NEED_S3
TF_NEED_VERBS=0; export TF_NEED_VERBS
TF_SET_ANDROID_WORKSPACE=0; export TF_SET_ANDROID_WORKSPACE
```
Then, I ran:
```console
$ ./configure
$ bazel --nohome_rc --nosystem_rc build --color=no --jobs=4 --config=opt --config=noaws --config=nogcp --config=nohdfs --config=noignite --config=nokafka --config=nonccl --config=v2 //tensorflow/tools/pip_package:build_pip_package
```
The bug seems to occur regardless of whether or not `--config=v2` is added, as that is the default. Using `--config=nogcp --config=v1` seems to work, and not using `--config=nogcp` also seems to work. Is GCP support required in v2?

**Any other info / logs**

[build log](https://github.com/tensorflow/tensorflow/files/3928307/spack-build-out.txt)
[build environment](https://github.com/tensorflow/tensorflow/files/3928308/spack-build-env.txt)

This work is being done in conjunction with https://github.com/spack/spack/pull/13112

I've seen #25644 #24447 #11807 #14522 and other similar issues, but they look specific to CUDA/NCCL or are due to older versions of Bazel.

**Comment by adamjstewart** [comment]
This bug is no longer present in 2.1.0. I'll close this, but if anyone knows which commit fixed this, I can patch 2.0.0.

**Comment by adamjstewart** [comment]
For the record, this bug seems to be present in 2.0.0 and 2.0.1, and seems to have been fixed by the following commit:
```
commit 837c8b6b35429ec99d977c609fe65c0e79f4569a
Author: Gunhan Gulsoy <gunan@google.com>
Date:   Wed Sep 4 13:43:49 2019 -0700

    Remove contrib cloud bigtable and storage ops/kernels.
    Thse have moved to github.com/tensorflow/io
    PiperOrigin-RevId: 267222179
```
